web-client/src/app.jsx

```diff
@@ -154,18 +152,36 @@ const app = {
       return value;
     });
 
-    const token =
-      (await applicationContext
-        .getUseCases()
-        .getItemInteractor(applicationContext, { key: 'token' })) ||
-      presenter.state.token;
-    presenter.state.token = token;
-    applicationContext.setCurrentUserToken(token);
+    if (process.env.IS_LOCAL) {
+      const token =
+        (await applicationContext
+          .getUseCases()
+          .getItemInteractor(applicationContext, { key: 'token' })) ||
+        presenter.state.token;
+      presenter.state.token = token;
+      applicationContext.setCurrentUserToken(token);
+    }
 
     presenter.state.cognitoLoginUrl = applicationContext.getCognitoLoginUrl();
 
     presenter.state.constants = applicationContext.getConstants();
 
+    // we must do all this logic before we start the router or render react
+    let startRefreshSequence = false;
+    const isNewTabOpened = !window.location.href.includes('?code');
```

Comment:
How do we know this means a new tab was opened? I have never seen this before!

Reply:
I'm just assuming.   There are only two ways to get into our private application.  Either you logged in via cognito (which means you were redirected to our application to `/log-in?code=ABC`, or you tried to load the application via a bookmark, new tab, or link.   Maybe I should change the variable name?